storybook/src/pages/amsterdam/common/AppHeader.tsx

```diff
@@ -1,13 +1,74 @@
-import { Grid, Header, PageMenu } from '@amsterdam/design-system-react'
-import { Alignment as PageMenuStory } from '../../../components/PageMenu/PageMenu.stories'
-
-export const AppHeader = () => (
-  <Grid>
-    <Grid.Cell span="all">
-      <Header
-        links={<PageMenu alignEnd>{PageMenuStory.args?.children}</PageMenu>}
-        menu={<button className="ams-header__menu-button">Menu</button>}
-      />
-    </Grid.Cell>
-  </Grid>
-)
+/**
+ * @license EUPL-1.2+
+ * Copyright Gemeente Amsterdam
+ */
+
+import { Grid, Header, PageMenu, SearchField } from '@amsterdam/design-system-react'
+import { MenuIcon, SearchIcon } from '@amsterdam/design-system-react-icons'
+import { useMemo, useState } from 'react'
+import { Default as MegaMenuStory } from '../../../components/MegaMenu/MegaMenu.stories'
+
+type Feature = 'menu' | 'search'
+
+export const AppHeader = () => {
+  const [visibleFeature, setVisibleFeature] = useState<Feature | undefined>(undefined)
+
+  const toggleFeature = (feature: Feature) =>
+    setVisibleFeature((currentFeature) => (feature === currentFeature ? undefined : feature))
+
+  const visibilityClass = (feature: Feature) => (feature === visibleFeature ? null : 'ams-visually-hidden')
+
+  const HeaderMenu = useMemo(
+    () => (
+      <PageMenu alignEnd>
+        <PageMenu.Item rank="secondary">
+          <PageMenu.Link href="#">English</PageMenu.Link>
+        </PageMenu.Item>
+        <PageMenu.Item rank="secondary">
+          <PageMenu.Link href="#">Contact</PageMenu.Link>
+        </PageMenu.Item>
+        <PageMenu.Item>
+          <PageMenu.Button icon={SearchIcon} onClick={() => toggleFeature('search')}>
+            Zoeken
+          </PageMenu.Button>
+        </PageMenu.Item>
+        <PageMenu.Item>
+          <PageMenu.Button icon={MenuIcon} onClick={() => toggleFeature('menu')}>
+            Menu
+          </PageMenu.Button>
+        </PageMenu.Item>
+      </PageMenu>
+    ),
+    [toggleFeature],
+  )
+
+  const Menu = () => (
+    <Grid as="nav" className={visibilityClass('menu')} paddingBottom="large">
```

Comment:
Getting some ts errors here, no `nav` in `as` property and possible `null` for `className`.

Reply:
The first should be cache; see `Grid.tsx`. Fixed the second.